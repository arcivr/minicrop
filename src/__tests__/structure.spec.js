const Structure = require('../structure.js')

describe("build", () => {
  const { build } = Structure.default
  var root

  beforeEach(() => {
    document.body.innerHTML = "<div class='minicrop' data-image='image'></div>"
    root = document.getElementsByClassName('minicrop')[0]
  })

  test('snapshot', () => {
    expect(build(root))
      .toMatchSnapshot()
  })

  test('includes image', () => {
    expect(build(root).innerHTML)
      .toMatch('src="image"')
  })

  describe("ratio", () => {
    test('defaults to 1:1 ratio', () => {
      expect(build(root).style.paddingTop)
        .toEqual("100%")
    })

    test('defaults to 1:1 ratio', () => {
      root.dataset.ratio = 0.54321

      expect(build(root).style.paddingTop)
        .toEqual("54.321%")
    })
  })

  describe("zoom", () => {
    test('defaults to 100% zoom', () => {
      expect(build(root).innerHTML)
        .toMatch('height="100%"')
    })

    test('includes zoom level', () => {
      root.dataset.zoom = "50%"

      expect(build(root).innerHTML)
        .toMatch('height="50%"')
    })
  })
})
