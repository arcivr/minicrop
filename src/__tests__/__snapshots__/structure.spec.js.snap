// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build snapshot 1`] = `
<div
  class="minicrop"
  data-image="image"
>
  
    
  <img
    class="image"
    height="100%"
    src="image"
  />
  
    
  <div
    class="crop"
  >
    
      
    <!-- &lt;img src="image" class="image-preview" height="100%" /&gt; -->
    
    
  </div>
  
  
</div>
`;
